Fix stdatomic.h _Atomic keyword in C++ mode (GCC 13.x)

WebKit bmalloc forces .c files to compile as C++ (-x c++), but GCC's
<stdatomic.h> uses the _Atomic keyword which is only valid in C11/C23,
not in C++. Wrap the include to use <atomic> in C++ mode.

--- a/Source/bmalloc/libpas/src/libpas/pas_utils.h
+++ b/Source/bmalloc/libpas/src/libpas/pas_utils.h
@@ -42,7 +42,17 @@
 #else
 #define PAS_HAVE_PTHREAD_MACHDEP_H 0
 #endif
 
 #include <limits.h>
-#include <stdatomic.h>
+
+#if defined(__cplusplus)
+#include <atomic>
+using std::atomic_bool;
+using std::atomic_int;
+using std::atomic_uint;
+using std::atomic_long;
+using std::atomic_ulong;
+using std::atomic_ullong;
+#else
+#include <stdatomic.h>
+#endif
 #include <stdbool.h>
 #include <stdint.h>
 #include <string.h>
